### ISSUE TITLE
GitSavvy commit: Pull from history (enhancement request)

### ISSUE THREAD

**Issue by tomngo** [timbrel/GitSavvy#487]
I frequently pull bits of information from previous commit entries, especially the most recent one. Can we make it easier to do this from inside a GitSavvy commit session? Right now, my workaround is to navigate to the Rebase dashboard.

Possibilities:
- List prior commit messages in the comments.
- Make a keystroke to pull in the most recent commit message.
- Make a keystroke to open a separate window with commit messages listed last first.

**Comment by jimpriest** [comment]
Old request but I would find this useful as well :)

**Comment by kaste** [comment]
Is that only for the commit subject, or the complete message? Only the last commit (or actually current HEAD)?

**Comment by jimpriest** [comment]
For me just the last commit subject would be fine.

**Comment by kaste** [comment]
I'll think there is a duplicate issue, have to look up later.

This would generally be useful if you e.g make fixup or squash commits where you reference the sha or subject of a commit.

The typical workflow within GitSavvy would be to open a short log upon a keystroke where you select a specific commit. After that we would insert the subject at the current cursor.

This should be a straight forward addition if we can agree on a good key bindings.

**Comment by kaste** [comment]
Duplicate of #89

**Comment by tomngo** [comment]
It's nice to have authored a duplicate of such a low-numbered issue. ;-)

I'll comment there, but in my case it would be helpful to have access to the complete message, and to be able to browse arbitrarily far in the past.

**Comment by kaste** [comment]
The discussion should stay here bc the other is closed and went off topic anyway.

You probably won't reuse the whole message so it's more for reading and maybe copy pasting bits of it? It's maybe like opening the log/graph view via shortcut then.

**Comment by jimpriest** [comment]
Yep - I'm lazy and looking for easy way to copy/paste bits of previous message. :)

**Comment by kaste** [comment]
@tomngo If you need the complete message and "arbitrary" far in the past, the starting point would be probably the the log panel.  E.g. add 

```
  {
      "keys": ["ctrl+l"],
      "command": "gs_log_current_branch",
      "context": [
          { "key": "setting.command_mode", "operator": "equal", "operand": false },
          { "key": "setting.git_savvy.commit_view", "operator": "equal", "operand": true }
      ]
  },
```

to your `Default.sublime-keymap`. Then on `ctrl+l` a menu/panel opens showing the git log. In addition, the view "splits" t.i. an output panel opens showing the content of the commit. 

That's almost there I think, so it would be interesting what fine tuning we would need here to make it better. 

(The variation, copy and paste the commit subject at the cursor, would be a simple addition around this workflow. What else?)

**Comment by tomngo** [comment]
My use is very similar to @jimpriest 's . I'd that it's beyond laziness: it starts there, but it's also sometimes about keeping a common format for similar commits.

I like the idea of starting with the log panel. Will try.

**Comment by asfaltboy** [comment]
With regards to common commit message format, a somewhat related feature request is #361

**Comment by kaste** [comment]
The general workflow with `ctrl+l` works nice and already helped me a lot. We should change the menu though because maybe not all options are that useful.

- It would be nice if I could jump to the commit detail panel just using the keyboard. Maybe `<TAB>` to jump from the "menu"/list of commits to the panel. Over there I could just look around, and copy paste whatever I want. 

- I.e. if I already typed `fixup!` or (`squash!`) it could insert the commit subject of the selected commit for me. Right now, I select the commit I want, then go down select copy full sha. (It is better to use the commit message though because it doesn't change if I later rebase multiple times.)

We need to compile a set of useful commands. 

- Copy subject
- Copy sha

It might be okay to not have a sub menu here. Since the detail panel opens I already have access to all the information and I could copy paste.  

Currently and for sure(?) not useful: cherry pick, checkout

**Comment by kaste** [comment]
Just shipped with 2.27.0  I think that's a good step forward, `<tab>` on the very first char (basically before starting to type the message) will show the most recent commits.  `fixup<tab>` and `squash<tab>` will also show the log and generate a commit message suitable for "git --interactive".
